app/services/planting_predictions.rb

```diff
@@ -29,27 +29,50 @@ def percentage_grown
   end
 
   def start_to_finish_diff
-    (@planting.finished_at - @planting.planted_at).to_i if @planting.finished_at && @planting.planted_at
+    return unless @planting.finished_at && @planting.planted_at
+    (@planting.finished_at.to_datetime - @planting.planted_at.to_datetime).to_i
   end
 
-  def predict_days_before_maturity
+  def time_to_first_harvest
+    return unless @planting.harvested_at && @planting.planted_at
+    (@planting.harvested_at.to_datetime - @planting.planted_at.to_datetime).to_i
+  end
+
+  def predict_days_before_finished
     # calculate the number of days, from planted_at, until maturity
     if @planting.planted_at && @planting.finished_at
       start_to_finish_diff
     elsif @planting.crop_id
       plantings = other_finished_plantings_same_crop
-      PlantingPredictions.mean_days_until_maturity(plantings)
+      PlantingPredictions.mean_days_before_finished(plantings)
     end
   end
 
-  def self.mean_days_until_maturity(plantings)
+  def self.mean_days_before_finished(plantings)
     ## Given a set of finished plantings, calculate the average/mean time from start to finish
     differences = plantings.collect(&:start_to_finish_diff)
     differences.compact.sum / differences.compact.size unless differences.compact.empty?
   end
 
+  def self.mean_days_before_first_harvest(plantings)
+    ## Given a set of finished plantings, calculate the average/mean time from start to finish
+    differences = plantings.collect(&:time_to_first_harvest)
+    differences.compact.sum / differences.compact.size unless differences.compact.empty?
+  end
+
+  def predict_harvest
+    days = PlantingPredictions.mean_days_before_first_harvest(other_harvested_plantings_same_crop)
```

Comment:
You mean we're not training a convolutional neural network-based deep learning model on latitude, shade, pixel-wise analysis of photos, and the phase of the moon? Awwwwwww.